File: Frontend/angular-ecommerce/src/app/common/purchase.ts
import { Order } from './order';
import { Address } from './address';
import { Customer } from './customer';
import { OrdrItem } from './ordr-item';
export class Purchase {
 
    Customer?:Customer;
     shippingAddress?:Address;
     billingAddress?:Address;
     order?:Order;
     orderItems?:OrdrItem[]; 
}

File: BackEnd/src/main/java/com/example/ecommercebackend/dto/Purchase.java
package com.example.ecommercebackend.dto;

import com.example.ecommercebackend.Entity.Address;
import com.example.ecommercebackend.Entity.Customer;
import com.example.ecommercebackend.Entity.Order;
import com.example.ecommercebackend.Entity.OrderItem;
import lombok.Data;

import java.util.Set;

@Data
public class Purchase {
  private Customer customer;
  private Address shippingAddress;
  private Address billingAddress;
  private Order order;
  private Set<OrderItem> orderItems;

}
